pytorch_lightning/metrics/metric.py

```diff
@@ -265,3 +260,33 @@ def __setstate__(self, state):
         self.__dict__.update(state)
         self.update = self._wrap_update(self.update)
         self.compute = self._wrap_compute(self.compute)
+
+    def _apply(self, fn):
+        """ Overwrite _apply function such that we can also move metric states
+            to the correct divice when `.to`, `.cuda` ect methods are called
+        """
+        self = super()._apply(fn)
+        # Also apply fn to metric states
+        for key in self._defaults.keys():
+            current_val = getattr(self, key, None)
+            if current_val is not None and isinstance(current_val, torch.Tensor):
+                setattr(self, key, fn(current_val))
+            else:
+                setattr(self, key, [fn(cur_v) for cur_v in current_val])
```

Comment:
```suggestion
            elif is instance(current_val, Sequence):
                setattr(self, key, [fn(cur_v) for cur_v in current_val])
            else:
                raise TypeError
```

Also it is not clear, what happens if `current_val` is None, since that way the `else` block is executed, but `None` is not iterable.

Reply:
Good catch 👍